### DISCUSSION TITLE
filtering with long text not working when listing

### DISCUSSION THREAD

**Discussion by fehmi** [pocketbase/pocketbase#1101] [Q&A] [answered]
Example URL:

http://127.0.0.1:8090/api/collections/playground/records/?filter=(content='eyJ0aXRsZSI6IkhlbGxvIFdvcmxkISIsImZpbGVzIjpbeyJmaWxlbmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPGh0bWw+XG4gIDxoZWFkPlxuICAgIDx0aXRsZT5zaW1wbHkuanMgLSBIZWxsbyBXb3JsZCE8L3RpdGxlPlxuICA8L2hlYWQ+XG4gIDxib2R5PlxuICAgIDxoZWxsby13b3JsZD48L2hlbGxvLXdvcmxkPlxuICAgIDxzY3JpcHQgc3JjPVwiaHR0cHM6Ly9yb290L3NpbXBseS9zaW1wbHkuanNcIj48L3NjcmlwdD5cbiAgICA8c2NyaXB0PlxuICAgICAgZ2V0KFwiaGVsbG8td29ybGQuaHRtbFwiKTtcbiAgICA8L3NjcmlwdD5cbiAgPC9ib2R5PlxuPC9odG1sPiIsInNlbGVjdGVkIjpmYWxzZX0seyJmaWxlbmFtZSI6ImhlbGxvLXdvcmxkLmh0bWwiLCJjb250ZW50IjoiPHRlbXBsYXRlPlxuICA8aDE+SGVsbG8gV29ybGQhPC9oMT5cbiAgPHA+R3JlZXRpbmdzIGZyb20ge2RhdGEubmFtZX08L3A+IFxuPC90ZW1wbGF0ZT5cblxuPHN0eWxlPlxuICBoMSB7XG4gICAgZm9udC1mYW1pbHk6IEFyaWFsLCBIZWx2ZXRpY2E7XG4gIFx0Y29sb3I6IGJsdWU7XG4gIH1cbjwvc3R5bGU+XG5cbjxzY3JpcHQ+XG4gIGNsYXNzIHtcbiAgICBkYXRhID0ge1xuICAgICAgbmFtZTogXCJzaW1wbHkuanNcIlxuICAgIH1cbiAgfVxuPC9zY3JpcHQ+Iiwic2VsZWN0ZWQiOmZhbHNlfSx7ImZpbGVuYW1lIjoiY29tcG9uZW50LTEuaHRtbCIsImNvbnRlbnQiOiI8dGVtcGxhdGU+XG5cbjwvdGVtcGxhdGU+XG5cbjxzdHlsZT5cbjo6aG9zdCB7XG5cbn1cbjwvc3R5bGU+XG5cbjxzY3JpcHQ+XG5jbGFzcyB7XG5cbn1cbjwvc2NyaXB0PiIsInNlbGVjdGVkIjp0cnVlfV19')

Returns zero record. But I know there is a record for it. 

Do you know why and how can I fix?

**Comment by ganigeorgiev** [comment, marked as answer]
Make sure that:

- your request satisfies the List API rule of the collection
- your filter is properly url encoded
- that there is a record with `content` matching exactly the provided value (if you want a SQL `LIKE` behavior you can use the `~` operator)

**Reply by ganigeorgiev** [reply]
@fehmi I've tested it both locally and in the live demo and it works fine, so most like you don't have List API rule permissions or you need to properly encode the url (the SDKs will do that for you):

```
http://127.0.0.1:8090/api/collections/playground/records?filter=(content%3D%22eyJ0aXRsZSI6IkhlbGxvIFdvcmxkISIsImZpbGVzIjpbeyJmaWxlbmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPGh0bWw%2BXG4gIDxoZWFkPlxuICAgIDx0aXRsZT5zaW1wbHkuanMgLSBIZWxsbyBXb3JsZCE8L3RpdGxlPlxuICA8L2hlYWQ%2BXG4gIDxib2R5PlxuICAgIDxoZWxsby13b3JsZD48L2hlbGxvLXdvcmxkPlxuICAgIDxzY3JpcHQgc3JjPVwiaHR0cHM6Ly9yb290L3NpbXBseS9zaW1wbHkuanNcIj48L3NjcmlwdD5cbiAgICA8c2NyaXB0PlxuICAgICAgZ2V0KFwiaGVsbG8td29ybGQuaHRtbFwiKTtcbiAgICA8L3NjcmlwdD5cbiAgPC9ib2R5PlxuPC9odG1sPiIsInNlbGVjdGVkIjpmYWxzZX0seyJmaWxlbmFtZSI6ImhlbGxvLXdvcmxkLmh0bWwiLCJjb250ZW50IjoiPHRlbXBsYXRlPlxuICA8aDE%2BSGVsbG8gV29ybGQhPC9oMT5cbiAgPHA%2BR3JlZXRpbmdzIGZyb20ge2RhdGEubmFtZX08L3A%2BIFxuPC90ZW1wbGF0ZT5cblxuPHN0eWxlPlxuICBoMSB7XG4gICAgZm9udC1mYW1pbHk6IEFyaWFsLCBIZWx2ZXRpY2E7XG4gIFx0Y29sb3I6IGJsdWU7XG4gIH1cbjwvc3R5bGU%2BXG5cbjxzY3JpcHQ%2BXG4gIGNsYXNzIHtcbiAgICBkYXRhID0ge1xuICAgICAgbmFtZTogXCJzaW1wbHkuanNcIlxuICAgIH1cbiAgfVxuPC9zY3JpcHQ%2BIiwic2VsZWN0ZWQiOmZhbHNlfSx7ImZpbGVuYW1lIjoiY29tcG9uZW50LTEuaHRtbCIsImNvbnRlbnQiOiI8dGVtcGxhdGU%2BXG5cbjwvdGVtcGxhdGU%2BXG5cbjxzdHlsZT5cbjo6aG9zdCB7XG5cbn1cbjwvc3R5bGU%2BXG5cbjxzY3JpcHQ%2BXG5jbGFzcyB7XG5cbn1cbjwvc2NyaXB0PiIsInNlbGVjdGVkIjp0cnVlfV19%22)
```

**Reply by fehmi** [reply]
Hmm I see. Thank you. I’m using PHP to call pure endpoint without SDK. I will try to encode the string and let you know.

**Reply by fehmi** [reply]
Looks it was an encoding issue, like you said. When I encode the part between `...(content=` and `)` like below, problem solved:

`$record["response"] = file_get_contents("http://127.0.0.1:8090/api/collections/playground/records/?filter=(content=".	urlencode("'".$string."'").")");`

Thank you for your support and the great tool!

**Reply by ganigeorgiev** [reply]
@fehmi Please note that the above will encode only part of the query params and ideally you should encode everything (including the `content=` part). You can let PHP to do the encoding for you by using `http_build_query`, eg:

```php
$record["response"] = file_get_contents("http://127.0.0.1:8090/api/collections/playground/records?" . http_build_query(['filter' => "content='$string'"]);
```

**Reply by fehmi** [reply]
Yeah, looks much better.
